### PR TITLE
fix(design-system): button styling

### PR DESCRIPTION
**Pull request by mattwang44**
## Types of changes

* **Bugfix**

## Description

Fix button style based on [design spec](https://www.figma.com/file/W3hdepQ7v7ZGYRP6QctBXH/PyCon-Taiwan-%E5%AE%98%E7%B6%B2%E8%A6%96%E8%A6%BA-2022?node-id=0%3A1) (The `loading` style is not included in this PR). The following is the screenshot of the button designs.

image

## Steps to Test This Pull Request

Test with the following snippet

<details>
    <summary>click</summary>

```html
<div class="flex flex-row">
    <div class="flex flex-wrap flex-col mt-10 ml-10">
        <text-button href="link" class="pb-10" large>
            Large Primary
        </text-button>
        <text-button class="pb-10" large> Large Primary</text-button>
    </div>
    <div class="flex flex-wrap flex-col mt-10 ml-10">
        <text-button href="link" class="pb-10">
            Medium Primary
        </text-button>
        <text-button class="pb-10"> Medium Primary </text-button>
    </div>
    <div class="flex flex-wrap flex-col mt-10 ml-10">
        <text-button href="link" class="pb-10" small>
            Small Primary
        </text-button>
        <text-button class="pb-10" small> Small Primary </text-button>
    </div>
</div>
<div class="flex flex-row">
    <div class="flex flex-wrap flex-col mt-10 ml-10">
        <text-button href="link" class="pb-10" large secondary>
            Large Secondary
        </text-button>
        <text-button class="pb-10" large secondary>
            Large Secondary</text-button
        >
    </div>
    <div class="flex flex-wrap flex-col mt-10 ml-10">
        <text-button href="link" class="pb-10" secondary>
            Medium Secondary
        </text-button>
        <text-button class="pb-10" secondary>
            Medium Secondary</text-button
        >
    </div>
</div>
```
</details>

## Expected behavior

image

### CODE REVIEW

File: components/core/buttons/TextButton.vue [hunk 1]
```diff
@@ -88,67 +90,154 @@ export default {
...
+        font-size: 24px;
+        padding: 16px 48px;
+        min-width: 108px;
+    }
+
+    &.--small:not(.--is-link),
+    &.--small > a {
+        @apply h-12 font-semibold;
+        padding: 12px 36px;
+        min-width: 80px;
```
[the remark this answers is not in the document]
Comment (mattwang44) [reply components/core/buttons/TextButton.vue]: added a class `text-base` in [411382f](https://github.com/pycontw/pycontw-frontend/pull/214/commits/411382feffcbd76f6548bf86eb8ea4b1355450d3)

File: components/core/buttons/TextButton.vue [hunk 2]
```diff
@@ -88,67 +90,154 @@ export default {
...
-    min-width: 80px;
-    border-width: 3px;
+.core-button.--uppercase > a {
+    @apply uppercase;
 }
 
 .core-button.--rounded:not(.--is-link),
 .core-button.--rounded.--is-link > a {
     border-radius: 40px;
 }
```
[the remark this answers is not in the document]
Comment (mattwang44) [reply components/core/buttons/TextButton.vue]: delete the props of `rounded` in [411382f](https://github.com/pycontw/pycontw-frontend/pull/214/commits/411382feffcbd76f6548bf86eb8ea4b1355450d3)

**Review by lingin1230 — COMMENTED** [review]
請問這邊是暫時在等待翻譯嗎？

**Comment by lingin1230** [comment]
> 請問這邊是暫時在等待翻譯嗎？

File: i18n/registration/financial-aid.i18n.js [hunk 3]
```diff
@@ -3,20 +3,12 @@ import { genI18nMessages } from '~/utils/i18n.utils'
...
-                'While we hope to support everyone to participate in the community, the budget is unfortunately limited, ' +
-                'and so is the financial aid program. In cases of insufficient resources, ' +
-                'we wish to support those who are most in need and to maximize community gain available from the applicant’s participation.' +
-                'As such, we may be forced to turn down your application to prioritize others.',
-        ],
+        intro:
+            '財務補助方案採取審核制，所以請有需求的朋友主動與我們聯絡，' +
+            '我們收到所有申請後會根據預算及申請者狀況等因素，決定是否給予補助以及其資源多寡。' +
+            '由於會議經費有限，因此能補助的數量不多；' +
+            '這種情況下我們希望受理的對象大致為最需要被幫助的人或對與會者幫助最大的人。' +
```
Comment (lingin1230) [inline i18n/registration/financial-aid.i18n.js:10]: 請問這邊是暫時在等待翻譯嗎？

**Review by lingin1230 — APPROVED** [review]
LGTM

**Pull request merged**